src/xSTIR/pSTIR/STIR.py

```diff
@@ -507,8 +512,9 @@ def allocate(self, value=0, **kwargs):
                 out.fill(numpy.random.random_sample(shape))
             elif value == 'random_int':
                 max_value = kwargs.get('max_value', 100)
-                out.fill(numpy.random.randint(
-                    max_value, size=shape))
+                out.fill(numpy.random.randint(max_value,size=shape))
+        elif value is None:
+            out = self.get_uniform_copy(0)
```

Comment:
if you just want to allocate memory, and `self` is not an empty template (`self.is_empty()` is `False`), then you can do `out = self.clone()`.

If your branch `DataContainer` does not have `is_empty()`, merge master.

Reply:
if self is a template would `get_uniform_copy` return a `DataContainer`?